### ISSUE TITLE
[BUG] inconsistant extrusion with arcs (G2/G3)

### ISSUE THREAD

**Issue by rerickson1** [MarlinFirmware/Marlin#23625]
### Did you test the latest `bugfix-2.0.x` code?

Yes, and the problem still exists.

### Bug Description

Starting with release 2.0.9.2 when printing with arcs, extrusion is inconsistant.
At the end of the seam, it will under extrude or over extrude.
This issue didn't happen in 2.0.9.1. 

### Bug Timeline

2.0.9.2

### Expected behavior

print quality with arcs should match or exceed print quality without arcs.

### Actual behavior

print quality with arcs is sub-par.

### Steps to Reproduce

Slice with Prusa slicer and run through ArcWelder Lib.

### Version of Marlin Firmware

2.0.9.3

### Printer model

Anet A8 Plus with modifications

### Electronics

MKS SGen L V2

### Add-ons

Bondtech extruder with E3D V6 hotend

### Bed Leveling

UBL Bilinear mesh

### Your Slicer

Prusa Slicer

### Host Software

OctoPrint

### Additional information & file uploads

The firmware I am running is 2.0.9.3 with these changes: https://github.com/MarlinFirmware/Marlin/compare/2.0.9.3...rerickson1:rde/2.0.9.3_a8+_custom

Issue never happened with 2.0.9.1. Could this change have introduced the issue?
https://github.com/MarlinFirmware/Marlin/pull/22599

Pictures of the issues:

over and under extrusion at end of seam:
![arcs_underextrude](https://user-images.githubusercontent.com/4430317/151216363-0e194b97-1ea8-41a9-9318-fcf72c4df1fb.JPG)

over extrusion at end of seam:
![arcs_overextrude](https://user-images.githubusercontent.com/4430317/151216446-2d24576d-0842-4a48-a4ea-2e15f63cfdf2.JPG)

same print printed **without** arcs:
![without_arcs](https://user-images.githubusercontent.com/4430317/151216591-d2f78eca-93c6-48b5-ac39-b57643500a75.JPG)

**Comment by thinkyhead** [comment]
> Could this change have introduced the issue?

Try reverting that change to see if it helps.

**Comment by rerickson1** [comment]
> Try reverting that change to see if it helps.

Reverting the change didn't help. There is still an issue.

What's very weird about this issue is that it only manifests after the firmware has run for a while.
See this print. The print takes about 3.5 hours at 0.2 mm layer height. The issue manifested towards the end of the print.
![1](https://user-images.githubusercontent.com/4430317/151631498-b56f94da-2ebf-4814-af23-3ebb9cdb6ec5.jpeg)

After this print was done, I re-printed the exact gcode and the issue showed throughout the print:
![2](https://user-images.githubusercontent.com/4430317/151631816-c01db1cd-278c-400b-b2ac-9fcaa42b70ea.jpeg)

I reverted the commit and updated my printer with the new firmware. Of course this reset the printer so I was interested to see what would happen. The issue is much less prevalent, but is showing quality issues at the end of the seam.
![3](https://user-images.githubusercontent.com/4430317/151631925-d916791c-b469-42c4-8bda-644f5061aedf.jpeg)

Then right after that print was done, I started it again to see what the result would be after the printer has been running a while and the quality was worse.
![4](https://user-images.githubusercontent.com/4430317/151632002-89acf766-77ab-4b5d-8a4b-ab05e17849bc.jpeg)

As a reminder, If I do not print with arcs, these prints come out flawless.

Here's the the model and gcode that can produce the issue.
[model_and_gcode.zip](https://github.com/MarlinFirmware/Marlin/files/7962300/model_and_gcode.zip)

**Comment by rerickson1** [comment]
This issue does exist in 2.0.9.1 as well. Can anyone else reproduce this issue?

**Comment by DiskDoctor** [comment]
Do you get this on a different printer? You may be experiencing HEAT CREEP

**Comment by rerickson1** [comment]
> Do you get this on a different printer? You may be experiencing HEAT CREEP

If I print without arcs, this never happens. So I have to believe there are no hardware issues with the printer.

**Comment by thinkyhead** [comment]
It would be interesting to see with isolated `G2`/`G3` moves whether the extruder is moving the same amount of filament as it does when doing the same with `G1`. Be sure to turn off any "advance extrusion" in your slicer when testing any of this stuff. Some slicers try to account for filament spring and back-pressure, but they are not able to do this well in relation to arcs.

**Comment by rerickson1** [comment]
I believe I have found the source of my error buried in the slicer settings. I used PrusaSlicer 2.4.0 for the tests. For the initial testing I had `use_relative_e_distances = 1`.
I re-ran the tests now with PrusaSlicer 2.4.1 and set `use_relative_e_distances = 0`. The results look perfect and there are no longer quality issues at the seam.

![IMG_2464](https://user-images.githubusercontent.com/4430317/164475174-040787d3-15ca-4c56-99a0-afe392ace8c4.png)
